### ISSUE TITLE
Selectable cards are not announced with screen reader

### ISSUE THREAD

**Issue by nicolethoen** [patternfly/patternfly-react#6798]
Currently, there is nothing about selectable cards that indicates to screen readers that they are selectable or interactable.

Selectable cards likely need an aria `role=option` and need to be placed in a parent container with an aria `role=listbox`
If multiple cards are selectable, then the parents container with the listbox role should also have `aria-multiselectable=true` and `aria-activedescendant` set up to contain the id of the card which currently has focus.
The parent will also need an `aria-label` if the list of cards does not have a title, or can reference a visual title using `aria-labelledby`.

When a card is selected, the card should use the `aria-selected=true`

this fix needs to be applied to the card/Selectable and selected example, the card/Legacy selectable and selected example, the Card view demo, the Tabs/modal tabs demo, Primary-detail card view demo.

**Comment by jessiehuff** [comment]
This is definitely an issue and also exists similarly for Tile with [this issue](https://github.com/patternfly/patternfly-react/issues/5891). I'm not so sure on the `role=listbox` though since typically I've seen this role used in dropdown-like components. The other potential issue with using that role is that we would need to add all of the other [typical role expectations](https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox) like implementing up/down arrow keys instead of Tab. It's an interesting idea and worth a discussion, but I'm not positive on the approach.

**Comment by wise-king-sullyman** [comment]
@jessiehuff since `listbox` was the solution landed on in the Tile issue should that standard be followed here?

If not the only other approach I can think of would be to maybe make the cards `gridcells` and put them in a `grid`, but I think that's less optimal than `listbox`. If you have other suggestions though I'm all ears, my research so far hasn't led me to any.

**Comment by nicolethoen** [comment]
@jgiardino I was discussing this with @wise-king-sullyman this morning. 
I think the HTML markup for Cards and Tiles came long before the 'selectable' variants. So we'd likely need to re-evaluate the structure of the component and that'd be considered a breaking change. I will make a note to consider this more semantic approach when we get a change to make breaking changes.

**Comment by jgiardino** [comment]
Makes sense, @nicolethoen. Thanks for the update!
